File: internal/content/testdata/bench/typescript/nc-subsets-ii.ts
function subsets_with_dup(nums: number[]): number[][] {
    function cmpArr(a: number[], b: number[]): number {
        const n = Math.min(a.length, b.length);
        for (let i = 0; i < n; i++) {
            if (a[i] < b[i]) return -1;
            if (a[i] > b[i]) return 1;
        }
        return a.length - b.length;
    }

    const res: number[][] = [];
    nums.sort((a, b) => a - b);

    function backtrack(start: number, current: number[]): void {
        res.push([...current]);
        for (let i = start; i < nums.length; i++) {
            if (i > start && nums[i] === nums[i - 1]) continue;
            current.push(nums[i]);
            backtrack(i + 1, current);
            current.pop();
        }
    }

    backtrack(0, []);
    const sorted = res.map(s => [...s].sort((a, b) => a - b));
    sorted.sort(cmpArr);
    return sorted;
}

File: internal/content/testdata/bench/java/nc-subsets-ii.java
import java.util.*;

class Solution {
    public long[][] subsets_with_dup(long[] nums) {
        List<long[]> res = new ArrayList<>();
        Arrays.sort(nums);
        backtrack(nums, 0, new ArrayList<>(), res);
        // Sort result like Python: each subset sorted, then sort all subsets
        res.sort((a, b) -> {
            int len = Math.min(a.length, b.length);
            for (int i = 0; i < len; i++) {
                if (a[i] != b[i]) return Long.compare(a[i], b[i]);
            }
            return Integer.compare(a.length, b.length);
        });
        return res.toArray(new long[0][]);
    }

    private void backtrack(long[] nums, int start, List<Long> current, List<long[]> res) {
        long[] arr = new long[current.size()];
        for (int i = 0; i < current.size(); i++) arr[i] = current.get(i);
        Arrays.sort(arr);
        res.add(arr);
        for (int i = start; i < nums.length; i++) {
            if (i > start && nums[i] == nums[i - 1]) continue;
            current.add(nums[i]);
            backtrack(nums, i + 1, current, res);
            current.remove(current.size() - 1);
        }
    }
}
